language-plutus-core/src/Language/PlutusCore/CBOR.hs

```diff
@@ -0,0 +1,208 @@
+module Language.PlutusCore.CBOR ( encodeProgram
+                                , decodeProgram
+                                , readProgram
+                                , writeProgram
+                                ) where
+
+import           Codec.CBOR.Decoding
+import           Codec.CBOR.Encoding
+import           Codec.CBOR.Read
+import           Codec.CBOR.Write
+import           Control.Recursion
+import qualified Data.ByteString.Lazy           as BSL
+import           Language.PlutusCore.Lexer.Type
+import           Language.PlutusCore.Name
+import           Language.PlutusCore.Type
+import           PlutusPrelude
+
+encodeTyBuiltin :: TypeBuiltin -> Encoding
+encodeTyBuiltin bi =
+    let i = case bi of
+            TyByteString -> 0
+            TyInteger    -> 1
+            TySize       -> 2
+    in encodeTag i
+
+decodeTyBuiltin :: Decoder s TypeBuiltin
+decodeTyBuiltin = go =<< decodeTag
+    where go 0 = pure TyByteString
```

Comment:
I feel slightly sad that these correspondences are just repeated twice, but I don't have a good idea for how to represent a bijection in such a way that we get exhaustiveness checking...